templates/_macros/rich-vertical-list.jinja

```diff
@@ -0,0 +1,127 @@
+# Params
+# title_text (string) (required): Title of the rich vertical list
+# flipped (boolean) (optional): Whether the list items are flipped so image is on the left and the text is on the right. Defaults to false.
+# Slots
+# description: Paragraph-style description content
+# logo_section_image_[1-8]: Logo section image, assumed to be img.p-logo-section__logo
```

Comment:
I'm a bit hesitant about implementing it like this.

Having multiple separate slots seems a bit mundane, especially that it seems you are required to put a specific element with a specific class name it it. Similar with list item below.

We are making strong assumptions there. Pattern provides the wrapping element of a component (`p-logo-section` or `p-list-divided`) and we are expecting content with class names of child element of these specific components. So we assume that developer understands that a particular component (that they are not using directly) is there. 

I don't think we should be doing that. 

We probably need to either have one slot per component and ask for a full specific component:

```
   {% if slot=="logo-section" %}
    <div class="p-logo-section">… </div>
   {% endif %}
```

or not require HTML at all, and only ask for contents of the list items, without wrapping element or class name.

```
   {% if slot=="list_item_1" %}
    Contents of list item without the element itself
   {% endif %}
```

But this wont work in case of logo images, and would be limited to a specific type of list (it will not be possible to change type of list or bullets).

Reply:
Switched to using a single logo section slot.

Investigating best way to adjust the list items.

Reply:
@bartaz 

I'm hesitant to make users pass in the list as its own element (it seems rather unwieldy, since the point of this pattern *is* the list, I'd like to try to abstract its markup away slightly if possible). This would also prevent us from placing limits on the  total number of list items.

I also don't think we should restrict people to a specific type of list. If I had to implement it in one of these two ways, I'd probably choose the first way (passing the entire list in).

We could also pass in the list as an array of JSON objects, although this is a bit of a departure from our previous param/slot conventions:

```html
{% call(slot) rich_vertical_list(
  title_text='H2 - this can take up to two lines, but best to stick to one line.',
  list_items=[
    {
      'classes': ['is-ticked'],
      'html': 'No mandatory subscriptions for Ubuntu'
    },
    {
      'classes': ['is-ticked'],
      'html': (
        '<a href="https://ubuntu.com/security">Security</a>, '
        '<a href="https://ubuntu.com/support">Support</a>, '
        'and fully managed '
        '<a href="https://ubuntu.com/openstack/managed">Infra</a> '
        'and <a href="https://ubuntu.com/managed/apps">Apps</a>'
      ),
    },
    {
      'classes': ['is-ticked'],
      'html': 'Public cloud, Data centre, Edge cluster and Appliances',
    }
  ]
) -%}
  {% set list_items = list_items[:15] %}
  {%- if list_items|length > 0 %}
  <!--Optional list-->
  <ul class="p-list--divided">
    {% for list_item in list_items %}
      <div class="p-list__item {{ (list_item.classes or []) | join(' ') }}">
        {{ list_item.html | safe }}
      </div>
   {% endfor %}
 </ul>
{%- endif -%}
 
{% endcall %}
```

@pastelcyborg - any opinion here?

Reply:
Updated to passing in list item contents via slots, and setting list tick type using parameter.